libs/sdk-core/src/lib/db/search/agentic.ts

```diff
@@ -203,103 +203,21 @@ export namespace Agentic {
-      const keys = path.split('.');
-      return keys.reduce((acc, key) => {
-        if (!acc) {
-          return undefined;
-        }
-        if (key.startsWith('[')) {
-          key = key.replace('[', '').replace(']', '');
-          if (Array.isArray(acc)) {
-            try {
-              const index = parseInt(key, 10);
-              return acc[index];
-            } catch (e) {
-              return undefined;
-            }
-          } else {
-            if (acc[key]) {
-              return acc[key];
-            } else {
-              try {
-                const index = parseInt(key, 10);
-                return Object.values(acc)[index];
-              } catch (e) {
-                return undefined;
-              }
-            }
-          }
-        } else if (acc[key]) {
-          return acc[key];
-        } else {
-          return undefined;
-        }
-      }, obj as any);
-    }
-
-    private format(expression: string, skipMissing = true): string | undefined {
-      console.log('format', expression);
-      let missing = false;
-      const formatted = expression.replace(VARIABLES, (match, p1) => {
-        const value = this.getObjectValue(this.context, p1 as string);
-        if (value === undefined) {
-          missing = true;
-        }
-        return value !== undefined ? `${value}` : '';
-      });
-      return missing && skipMissing ? undefined : formatted;
-    }
-
-    private eval<T>(expression: string): T | undefined {
-      const expr = this.format(expression, false);
-      if (!expr) {
-        return undefined;
-      }
-      try {
-        return eval(expr) as T;
-      } catch (e) {
-        console.error(expr);
-        console.error(e);
-        return undefined;
-      }
-    }
-
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    private formatObject(obj: any): any {
-      if (typeof obj === 'string') {
-        return this.format(obj);
-      } else if (Array.isArray(obj)) {
-        const arr = obj.map((item) => this.formatObject(item)).filter((item) => item !== undefined);
-        return arr.length > 0 ? arr : undefined;
-      } else if (typeof obj === 'object') {
-        return Object.entries(obj).reduce((acc, [key, value]) => {
-          const formatted = this.formatObject(value);
-          acc[key] = formatted;
-
-          return acc;
-        }, obj);
-      } else {
-        return obj;
+      switch (step.action.type) {
+        case 'predict':
+          return this.runPredictAction(stepId, step.action);
+        case 'find':
+          return this.runFindAction(stepId, step.action);
+        case 'ask':
+          return this.runAskAction(stepId, step.action);
+        case 'web':
+          return this.runWebAction(stepId, step.action);
+        case 'api':
+          return this.runAPIAction(stepId, step.action);
+        case 'user':
+          return this.runUserAction(stepId, step.action);
+        default:
+          return of(false);
```

Comment:
your previous code was returning `of(true)` by default, did you change it on purpose?

Reply:
yes